rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperServer.java

```diff
@@ -68,11 +69,35 @@ public void run()
 
     public static void startServer(Configuration conf)
     {
+        try {
+            MetricsFactory.init(conf);
+        }
+        catch (Exception ex) {
+            log.error("Not able to initialize MetricsFactory ", ex);
+        }
+
         bookKeeper = new BookKeeper(conf);
         DiskMonitorService diskMonitorService = new DiskMonitorService(conf, bookKeeper);
         diskMonitorService.startAsync();
         processor = new BookKeeperService.Processor(bookKeeper);
         log.info("Starting BookKeeperServer on port " + getServerPort(conf));
+
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run()
+            {
+                String shutdownMsg = "Shutting down BookKeeperServer.";
+                log.info(shutdownMsg);
+                try {
+                    MetricsFactory.close();
+                }
+                catch (Exception e) {
+                    log.error("Error in closing MetricsFactory : " + e.getClass().getName() + " "
+                    + e.getMessage(), e);
```

Comment:
Do we have a standard for whether to use `String.format()` over concatenation like this?